### ISSUE TITLE
[HOLD for payment 2024-03-22] [$500] [Simplified Collect][Categories] CategorySettingsPage: rename category, delete category, onyx actions

### ISSUE THREAD

**Issue by luacmartins** [Expensify/App#35709]
Implement `CategorySettingsPage: rename category, delete category, onyx actions` from the [CategorySettingsPage: page init, enable/disable categories, display name, onyx action for enabling/disabling category](https://docs.google.com/document/d/1gk3xqOs7epMbUrSSiX8K7YcqfPLVgqEos0sf-D-GMDA/edit)

<details><summary>Issue Owner</summary>Current Issue Owner: @ArekChr</details>
<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~019c879178a6d4badc</li>
        <li>Upwork Job ID: 1755658765113217024</li>
        <li>Last Price Increase: 2024-02-08</li>
    </ul>
</details>

**Comment by luacmartins** [comment]
@ArekChr can you please comment here so I can assign you to the issue?

**Comment by ArekChr** [comment]
Sure, please assign me @luacmartins

**Comment by jeremy-croff** [comment]
Hi I'm trying to review this issue but do not have access to the document in the OP.
@luacmartins

**Comment by slafortune** [comment]
moving this to weekly while on hold for https://github.com/Expensify/App/issues/35705

**Comment by slafortune** [comment]
Sorry for the delay with this @jeremy-croff.  I've emailed you our NDA to sign to jeremycroff@gmail.com.  Once that is complete I can definitely share this doc with you 👍

**Comment by trjExpensify** [comment]
Hi @jeremy-croff, this issue is being worked on by an existing team of people - so sharing the doc with you isn't necessary in this case. You'll want to look for issues in the repo that have the `Help wanted` label. Thanks!

**Comment by luacmartins** [comment]
What @trjExpensify said. I imagine @akinwale might need access to review the PR and make sure we're following the implementation laid out in the doc though

**Comment by jeremy-croff** [comment]
> Hi @jeremy-croff, this issue is being worked on by an existing team of people - so sharing the doc with you isn't necessary in this case. You'll want to look for issues in the repo that have the `Help wanted` label. Thanks!

I understand, thanks. It had both External and Help wanted at one time. You can disregard my request to look into this ticket.

**Comment by trjExpensify** [comment]
All good, all good! 👍

**Comment by slafortune** [comment]
@ArekChr do you have an update here?

**Comment by ArekChr** [comment]
Hey, the [main category settings page](https://github.com/Expensify/App/issues/35708) is currently under review, which is needed for this feature. It will soon be merged, and then I'll create a draft PR

**Comment by slafortune** [comment]
Great!  Thanks for the details!

**Comment by ArekChr** [comment]
Hey there, soon I will create draft PR

**Comment by ArekChr** [comment]
@luacmartins Linked draft PR

**Comment by luacmartins** [comment]
Thank you!

**Comment by ArekChr** [comment]
PR ready for review

**Comment by slafortune** [comment]
@akinwale can you propose [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps to ensure the new feature will work correctly on production in further releases so I can get this paid today?

**Comment by akinwale** [comment]
> * [x]  [@akinwale] Please propose [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps to ensure  the new feature will work correctly on production in further releases.

**Regression Test Steps**
* Launch Expensify
* Log in as an the admin of a paid policy (Collect or Control) and navigate to the Workspace Settings > Select a workspace > Categories.
* Choose some categories that are currently disabled.
* Click the options menu in the top right corner and select "Enable Categories".
* Verify that the selected categories are now enabled.
* Choose some categories that are currently enabled.
* Click the options menu in the top right corner again and select "Disable Categories".
* Verify that these categories are now disabled.
* Choose any categories you wish to delete.
* In the top right corner, click the options menu and choose "Delete Categories".
* Verify that the selected categories were removed.

Do we agree 👍 or 👎?

**Comment by akinwale** [comment]
@slafortune Done!

**Comment by slafortune** [comment]
Offer sent @akinwale https://www.upwork.com/nx/wm/offer/101522153

**Comment by akinwale** [comment]
> Offer sent @akinwale https://www.upwork.com/nx/wm/offer/101522153

Accepted. Thanks!

**Comment by slafortune** [comment]
@akinwale  - Paid $500 via Upworks
@ArekChr does not require payment (Contractor)
